nomad/server.go

```diff
@@ -1653,6 +1659,42 @@ func (s *Server) setupSerf(conf *serf.Config, ch chan serf.Event, path string) (
 	return serf.Create(conf)
 }
 
+func (s *Server) maybeAutoBootstrapACLs() error {
+	if !s.config.ACLEnabled || s.config.ACLBootstrapToken == "" {
+		// Exit without bootstrapping the ACL system
+		return nil
+	}
+
+	resp := structs.ACLTokenUpsertResponse{}
+	err := s.RPC("ACL.Bootstrap",
+		&structs.ACLTokenBootstrapRequest{BootstrapSecret: s.config.ACLBootstrapToken},
+		&resp,
+	)
```

Comment:
I know we chatted about this a bit offline, but I'm thinking through this a bit more... because this function is blocking, we need to make an RPC call to the leader in order to start the server _at all_ (although I like that you're not returning an error on error at least). Ideally users will remove the token from the config, obviously, but I can definitely see a case where users don't, then add servers afterwards, and have a restart hang as a result.

I think we want to call `s.State().CanBootstrapACLToken()` here first, and only if that doesn't return an index should we try to bootstrap again. I don't think we really care about the reset index here?